cmd/expired-authz-purger/main.go

```diff
@@ -257,10 +257,28 @@ func deleteAuthorization(db eapDB, table, id string) error {
 	// Delete challenges + authorization. We delete challenges first and fail out
 	// if that doesn't succeed so that we don't ever orphan challenges which would
 	// require a relatively expensive join to then find.
-	_, err := db.Exec("DELETE FROM challenges WHERE authorizationID = ?", id)
+	// When deleting challenges, we first SELECT by authorizationID then delete
```

Comment:
Could you add some more here about why this works, it's pretty counterintuitive that this would help (also this comments implies this _does_ work, while the PR desc implies it _may_, has this been tested and shown to actually do anything different?).

Reply:
We haven't confirmed that this does anything different. Right now the symptoms are: Once every couple of weeks, a random `DELETE FROM challenges WHERE authorizationID="xyz";` query hangs for a very long time, evidently doing a full table scan (as evidenced by a very large number of examined rows). We don't have a reliable reproduction case, unfortunately. Most of the time, those DELETEs correctly use the `authorizationID` index on the challenges table, but occasionally they seem to fail to do so.

It's also worth noting that sometimes these hanging queries are on the master, and sometimes on a replica. In other words, sometimes we have a DELETE that worked fine on the master, but when the replica tries to execute it, it does a full table scan.

We can force index use with `FORCE INDEX`, but that only works for `SELECT` queries. However, I have higher confidence that if we do our DELETEs based on the challenge ID, they won't wind up doing a full table scan.

In the end, this is a speculative change, but I can't think of a better way to diagnose the problem, and at least this will probably do no harm.

Reply:
I'm +1 for trying to capture more of this context in the comment. I agree with @rolandshoemaker that the comment makes it sound like a definitive work-around.